lib/AST/ASTDumper.cpp

```diff
@@ -1473,12 +1473,15 @@ void Stmt::print(raw_ostream &OS, unsigned Indent) const {
 // Printing for Expr and all subclasses.
 //===----------------------------------------------------------------------===//
 
-static raw_ostream &operator<<(raw_ostream &os, AccessSemantics accessKind) {
+// Ideally, we'd like to define this as an operator (<<), but MSVC reports many
+// ambiguous resolution errors, as it tries to overload according to the enum's
+// underlying type, not primarily the actual enum.
```

Comment:
Okay, I have a much better fix that suddenly came to me when you commented. Basically, the problem was that the code referring to this `<<` operator is in an anonymous namespace. Moving `<<` into that anonymous namespace fixes this. Unfortunately, MSVC wasn't very helpful as it reported the error as an ambiguous overload error for `unsigned char`